express-notes-api/server.js

```diff
@@ -0,0 +1,90 @@
+import express from 'express';
+import { readFile, writeFile } from 'node:fs/promises';
+
+const app = express();
+
+const notesData = JSON.parse(await readFile('./data.json', { encoding: 'utf8' }));
```

Comment:
While reading the file data up front may save on execution time, it can result in out-of-date data issues if multiple requests come in from different users. To avoid that, pull updated data from the file with each request.